### PR TITLE
feat: support multiple create, edit, submit and review

### LINKED ISSUE #1: Multiple Review Accepts/Rejects in one tx

### LINKED ISSUE #2: Support adding one or more quests in quest chain creation flow

### LINKED ISSUE #4: Support multiple quest creations / edits in one tx.

### PR DESCRIPTION
**Pull request by parv3213**
- [x] SC changes
- [x] Existing test fix
- [x] Add new tests

### CODE REVIEW

File: contracts/interfaces/IQuestChain.sol [hunk 1]
```diff
@@ -9,25 +9,25 @@ interface IQuestChain {
     event QuestChainEdited(address indexed editor, string details);
     event QuestCreated(
```
Comment (dan13ram) [inline contracts/interfaces/IQuestChain.sol]: Should we rename the events to be "QuestsCreated" or "MultipleQuestsCreated" for clarity ?

Comment (dan13ram) [reply contracts/interfaces/IQuestChain.sol]: Hmm. We may then need to update the method names also.

Comment (parv3213) [reply contracts/interfaces/IQuestChain.sol]: Yeah, I couldn't make up my mind on the rename. Whatever you suggest. 
Alternate to `MultipleQuestsCreated ` can be just `QuestsCreated`.

File: contracts/QuestChain.sol [hunk 2]
```diff
@@ -191,58 +191,86 @@ contract QuestChain is
...
+            _reviewProof(_questorList[i], _questIdList[i], _successList[i]);
+        }
 
         emit QuestProofReviewed(
             _msgSender(),
-            _quester,
-            _questId,
-            _success,
-            _details
+            _questorList,
```
Comment (dan13ram) [inline contracts/QuestChain.sol]: `quester` and not `questor`

Comment (parv3213) [reply contracts/QuestChain.sol]: oops, fixing.

File: contracts/interfaces/IQuestChain.sol (hunk 1 above)
Comment (dan13ram) [reply contracts/interfaces/IQuestChain.sol]: @parv3213 sure lets go with `QuestsCreated` and also method names can be plural as well.
Would make it easier to differentiate in terms of old vs new abi also.

Comment (parv3213) [reply contracts/interfaces/IQuestChain.sol]: aye aye

File: contracts/QuestChain.sol (hunk 2 above)
Comment (vidvidvid) [reply contracts/QuestChain.sol]: we used to call them questors until we found out that
> questor (plural questores) A member of the clergy who provides forgiveness for sins.

:'D

**Comment by parv3213** [comment]
remove whenNotPaused from create and edit

**Comment by parv3213** [comment]
Good for review and merge.

**Review by dan13ram — APPROVED** [review]

**Pull request merged**